webapi/middleware.go

```diff
@@ -305,58 +305,92 @@ func vspAuth() gin.HandlerFunc {
 			return
 		}
 
-		// If the ticket was found in the database, we already know its
-		// commitment address. Otherwise we need to get it from the chain.
-		var commitmentAddress string
-		if ticketFound {
-			commitmentAddress = ticket.CommitmentAddress
-		} else {
-			dcrdClient := c.MustGet("DcrdClient").(*rpc.DcrdRPC)
+		altSigData, err := db.AltSigData(hash)
+		if err != nil {
+			log.Errorf("%s: db.AltSigData (ticketHash=%s): %v", funcName, hash, err)
+			sendError(errInternalError, c)
+			return
+		}
+		altSigAddr := ""
+		if altSigData != nil {
+			altSigAddr = altSigData.AltSigAddr
+		}
 
-			resp, err := dcrdClient.GetRawTransaction(hash)
-			if err != nil {
-				log.Errorf("%s: dcrd.GetRawTransaction for ticket failed (ticketHash=%s): %v", funcName, hash, err)
-				sendError(errInternalError, c)
-				return
+		// validate validates the signature in the request as signing
+		// the request body with private key corresponding to sAddr.
+		// Verification failure results in the client getting notified
+		// of the cause. Returns whether verification succeeded.
+		validate := func(kt bool) {
+			t := &ticket
+			validateSig := func(addr string) bool {
+				// Validate request signature to ensure ticket ownership.
+				err = validateSignature(reqBytes, addr, c)
+				if err != nil {
+					log.Warnf("%s: Bad signature (clientIP=%s, ticketHash=%s): %v", funcName, c.ClientIP(), hash, err)
```

Comment:
If I am reading correctly, a request which has an altSig in the database but has been signed with the commitment address will result in two log lines:

1. From failed altSig validation `Warn: Bad Signature`
1. From successful cmtAddr validation `Trace: Authed`

The warning seems unnecessary, nothing is wrong in this scenario. We only need a warning if both fail.

Reply:
Made so it doesn't fuss about the first error. Should I use more logic to maybe warn that both failed in case they do? It will only warn for the second one now.  95d7a0a